### ISSUE TITLE
Restaurant wifi Session says "HTTP request failed with status code: 0"

### ISSUE THREAD

**Issue by shovas** [oxen-io/session-ios#358]
**Describe the bug**

Using Session in a restaurant (Tim Hortons in Canada) and failed to send with error "HTTP request failed with status code: 0"

**To reproduce**

Send a message with Session in a restaurant (TIm Hortons).

**Screenshots or logs**

https://i.postimg.cc/kDP7b3VT/20210226-165728960-i-OS.jpg

**Smartphone (please complete the following information):**

 - Device: iPad
 - OS: Latest
 - Version of Session or latest commit hash: Latest

**Additional context**

N/A

**Comment by shovas** [comment]
I was wondering if session was hitting non-standard ports, anything besides 80 and 443?

**Comment by NoMoreNicksLeft** [comment]
I suspect that this is some system that blocks Session's network. My workplace does the same, and I get the same error message. If I turn off wifi, I can get it to work. I actually came here to github to suggest that the workaround/fix for it might be to include a feature that allows turning off wifi for session entirely and using cell as a fallback (iOS lets you turn off cell data per app, but not wifi data per app).

**Comment by shovas** [comment]
> I suspect that this is some system that blocks Session's network. My workplace does the same, and I get the same error message. If I turn off wifi, I can get it to work. I actually came here to github to suggest that the workaround/fix for it might be to include a feature that allows turning off wifi for session entirely and using cell as a fallback (iOS lets you turn off cell data per app, but not wifi data per app).

You think they're blocking Session-specific hosts or ips? That would be weird since Session isn't all that widely known.

But then again, it may speak to the need to decentralize the decentralization. Imho, all new apps should be literally peer-to-peer not pseudo p2p through "decentralized" nodes.

See iris.to for how they do that. You have to use coordinating nodes, but those can be run by everyone, but after that it's all p2p. Funny how napster and bittorrent have basically had it right this whole time.

**Comment by NoMoreNicksLeft** [comment]
I suspect that they're blocking hosts/IPs, at least where I am. I don't think they're aware of Session, but use someone else's service that just gives them a list of problematic hosts to stay away from. And those dirtbags seem to be aware of esoteric things quite often. That would account for how that restaurant is blocking it too. I doubt Tim Horton's IT staff are aware.

**Comment by GodGoldfish** [comment]
i think that this issue needs more investigation, it happened to me when i tried to share a media but not while texting iOS 15.2.1 & Session 1.12.1

**Comment by KeeJef** [comment]
Yeah Session will hit several high ports depending on the Service Nodes it communicates with, the default Service Node port is 22021 but it can be specified to be different by the Service Node operator. If the network firewall blocks high ports this may cause issues.
